### PR TITLE
Update strategy.md

### PR DESCRIPTION
**Pull request by gdennie**
I found the explanation of Strategy and Dependency injection near the top of the file confusing. After some additional reading elsewhere I thought this might be a bit clearer.

### CODE REVIEW

File: src/patterns/behavioural/strategy.md [hunk 1]
```diff
@@ -3,20 +3,26 @@
...
-particular problem, we define only the skeleton of the algorithm at an abstract
-level, and we separate the specific algorithm’s implementation into different
-parts.
-
-In this way, a client using the algorithm may choose a specific implementation,
-while the general algorithm workflow remains the same. In other words, the
-abstract specification of the class does not depend on the specific
-implementation of the derived class, but specific implementation must adhere to
-the abstract specification. This is why we call it "Dependency Inversion".
+The basic idea behind the Strategy pattern is actually the fundamental idea traits.
```
Comment (neithernut) [inline src/patterns/behavioural/strategy.md:10]:
```suggestion
The basic idea behind the Strategy pattern is actually the fundamental idea of traits.
```

File: src/patterns/behavioural/strategy.md [hunk 2]
```diff
@@ -3,20 +3,26 @@
...
+That is, an API or just algorithm can be fully defined for solving some particular problem 
+but have parts whose actual implementation is left abstract or overridable
```
Comment (neithernut) [inline src/patterns/behavioural/strategy.md:12]: This sentence doesn't read that nice. But the meaning does come across.

File: src/patterns/behavioural/strategy.md [hunk 3]
```diff
@@ -3,20 +3,26 @@
...
+
+It is also possible for these implementation pieces to be cemented in a type definition
+but a design can have them be provided through the API or invocation of the algorithm
+at runtime: the overridable parts are well defined since the abstract parts and their
+expectations are strongly typed, the requirements
+of the pieces are well defined and type checked at compile time.
+As a result, the API can be completely oblivious to the actual strategy being provided
+at runtime. As such, instead of only the caller being dependent on the 
+API, the API is now also dependent on the caller's provided strategy. 
+This is why we call it "Dependency Inversion".
```
Comment (neithernut) [inline src/patterns/behavioural/strategy.md:25]: This whole paragraph reads more convoluted than necessary.

File: src/patterns/behavioural/strategy.md [hunk 4]
```diff
@@ -98,17 +104,21 @@ fn main() {
...
 The main advantage is separation of concerns. For example, in this case `Report`
-does not know anything about specific implementations of `Json` and `Text`,
-whereas the output implementations does not care about how data is preprocessed,
-stored, and fetched. The only thing they have to know is a specific trait to
-implement and its method defining the concrete algorithm implementation
-processing the result, i.e., `Formatter` and `format(...)`.
+does not need to know anything about specific implementations of `Json` and `Text`
+while the output implementations does not need to care about how data is preprocessed,
+stored, and fetched. The only thing they have to know is a specific trait's API
+or algorithm invocation requirements, i.e., `Formatter` and `format(...)`.
```
Comment (neithernut) [inline src/patterns/behavioural/strategy.md:110]: This rewrite feels unnecessary.

**Review by neithernut — CHANGES_REQUESTED** [review]
Disclaimer: not a maintainer.

I do feel at least some of these changes are an improvement. But the writing does come across not as fluid as it could be.

**Comment by MarcoIeni** [comment]
Hey George, I'm sorry to see you closed this PR after Julian's feedback.
Is it because you didn't have the time to work on it?

**Comment by gdennie** [comment]
yes, unfortunately.

**Pull request closed without merging**